README.md
<p align="center">
<img src="./examples/favicon.ico" alt="mark text" width="100" height="100">
</p>

<h3 align="center">v-charts</h3>

<p align="center">
  <a href="https://travis-ci.org/ElemeFE/v-charts">
    <img src="https://travis-ci.org/ElemeFE/v-charts.svg?branch=master" alt="Build Status">
  </a>
  <a href="https://npmjs.org/package/v-charts">
    <img src="http://img.shields.io/npm/dm/v-charts.svg" alt="NPM downloads">
  </a>
  <a href="https://www.npmjs.org/package/v-charts">
    <img src="https://img.shields.io/npm/v/v-charts.svg" alt="Npm package">
  </a>
  <a>
    <img src="https://img.shields.io/badge/language-javascript-yellow.svg" alt="Language">
  </a>
  <a>
    <img src="https://img.shields.io/badge/license-MIT-000000.svg" alt="License">
  </a>
  <a href="https://gitter.im/ElemeFE/v-charts?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge">
    <img src="https://badges.gitter.im/ElemeFE/v-charts.svg" alt="Join the chat">
  </a>
</p>

<p align="center">
  <a href="http://frontend.tanzk.cn/teaching/tz-chart/#/">
    文档
  </a>
  <span> | </span>
  <a href="https://codesandbox.io/s/z69myovqzx">
    示例项目
  </a>
  <span> | </span>
  <a href="./README.md">
    English
  </a>
  <span> | </span>
  <a>
    中文
  </a>
</p>

> 基于 Vue2.x 封装的 Echarts 图表组件

## 特性

- **统一的数据格式：** 使用对前后端都友好的数据格式，方便生成和修改。
- **简化的配置项：** 通过简化的配置项，可以轻松实现复杂需求。
- **定制简单：** 提供多种自定义 Echarts 方式，可以方便的设置图表配置项。

## 支持性

支持所有现代浏览器及 IE10+ ，包括 pc 端和移动端。

## 安装

```
npm i @tzfe/v-charts echarts -S
```

## 快速上手

```html
<template>
  <div>
    <ve-line :data="chartData"></ve-line>
  </div>
</template>

<script>
  import VeLine from "v-charts/lib/line.common";
  export default {
    components: { VeLine },
    data() {
      return {
        chartData: {
          columns: ["日期", "销售量"],
          rows: [
            { 日期: "1月1日", 销售量: 123 },
            { 日期: "1月2日", 销售量: 1223 },
            { 日期: "1月3日", 销售量: 2123 },
            { 日期: "1月4日", 销售量: 4123 },
            { 日期: "1月5日", 销售量: 3123 },
            { 日期: "1月6日", 销售量: 7123 }
          ]
        }
      };
    }
  };
</script>
```

## 更新日志

每个版本的详细修改可以参考 [release notes](http://git.tanzk.cn/frontend/teaching/tz-chart.git/releases) 或者 [ChangeLog](./CHANGELOG_CN.md)。

## 贡献

在发起一个 pull request 之前，请先阅读[贡献指南](./CONTRIBUTING_CN.md)。

## License

[MIT](http://opensource.org/licenses/MIT)


CHANGELOG.md
# [1.0.0-beta.5](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.0.0-beta.4...v1.0.0-beta.5) (2020-04-25)



# [1.0.0-beta.5](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.0.0-beta.4...v1.0.0-beta.5) (2020-04-23)

# [1.0.0-beta.4](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.0.0-beta.3...v1.0.0-beta.4) (2020-04-22)

# [1.0.0-beta.3](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.0.0-beta.2...v1.0.0-beta.3) (2020-04-22)

# [1.0.0-beta.2](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.0.0-beta.1...v1.0.0-beta.2) (2020-04-22)

# [1.0.0-beta.1](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.1.1-alpha.1...v1.0.0-beta.1) (2020-04-15)

## [1.1.1-alpha.1](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.1.1-alpha.0...v1.1.1-alpha.1) (2020-04-13)

## [1.1.1-alpha.0](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.1.0-y.1...v1.1.1-alpha.0) (2020-04-11)

# [1.1.0-alpha.0](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.1.0-y.1...v1.1.0-alpha.0) (2020-04-11)

# [1.1.0-y.1](http://git.tanzk.cn/frontend/teaching/tz-chart/compare/v1.1.0-y.0...v1.1.0-y.1) (2020-04-08)

### Bug Fixes

- 去掉配置 ([24ed313](http://git.tanzk.cn/frontend/teaching/tz-chart/commits/24ed313ec26e902973bb38267b4dc607f9f64bc2))

### Features

- 文档修改 ([481c670](http://git.tanzk.cn/frontend/teaching/tz-chart/commits/481c67019d9ab9212e29072c15e9b513bf9d3705))
- 柱状图开发 ([e42303a](http://git.tanzk.cn/frontend/teaching/tz-chart/commits/e42303ac4bb4ded724b833ae8487a315da472767))
- pie 图开发 ([987157e](http://git.tanzk.cn/frontend/teaching/tz-chart/commits/987157efdd5dcd68098ed744cd66c13c2cdb7378))
- pie 图开发 ([4af1170](http://git.tanzk.cn/frontend/teaching/tz-chart/commits/4af1170b0b93e8cb47f5ccf2354ae02ac13ef4d1))


examples/data/line.js
export default {
  name: "折线图",
  type: "line",
  data: [
    // {
    //   name: "简单折线图",
    //   data: {
    //     columns: ["日期", "访问用户", "下单用户", "下单率"],
    //     rows: [
    //       { 日期: "1/1", 访问用户: 1393, 下单用户: 1093, 下单率: 0.32 },
    //       { 日期: "1/2", 访问用户: 3530, 下单用户: 3230, 下单率: 0.26 },
    //       { 日期: "1/3", 访问用户: 2923, 下单用户: 2623, 下单率: 0.76 },
    //       { 日期: "1/4", 访问用户: 1723, 下单用户: 1423, 下单率: 0.49 },
    //       { 日期: "1/5", 访问用户: 3792, 下单用户: 3492, 下单率: 0.323 },
    //       { 日期: "1/6", 访问用户: 4593, 下单用户: 4293, 下单率: 0.78 }
    //     ]
    //   },
    //   settings: {
    //     metrics: ["访问用户", "下单用户", "下单率"],
    //     dimension: ["日期"],
    //     columns: ["日期", "访问用户", "下单用户", "下单率"],
    //     rows: [
    //       { 日期: "1/1", 访问用户: 1393, 下单用户: 1093, 下单率: 0.32 },
    //       { 日期: "1/2", 访问用户: 3530, 下单用户: 3230, 下单率: 0.26 },
    //       { 日期: "1/3", 访问用户: 2923, 下单用户: 2623, 下单率: 0.76 },
    //       { 日期: "1/4", 访问用户: 1723, 下单用户: 1423, 下单率: 0.49 },
    //       { 日期: "1/5", 访问用户: 3792, 下单用户: 3492, 下单率: 0.323 },
    //       { 日期: "1/6", 访问用户: 4593, 下单用户: 4293, 下单率: 0.78 }
    //     ]
    //   },
    //   extend: {
    //     series: {
    //       markLine: {
    //         // 显示平均水平线
    //         silent: true, // 禁止事件
    //         symbol: "none", // 去掉箭头
    //         data: [{ type: "average", name: "平均值" }],
    //         label: {
    //           show: true,
    //           position: "end",
    //           formatter: function(p) {
    //             return Math.round(p.value);
    //           }
    //         },
    //         lineStyle: {
    //           color: "#F57535",
    //           width: 2,
    //           type: "dashed"
    //         }
    //       }
    //     }
    //   }
    // },
    // {
    //   name: "设置显示的指标维度",
    //   data: {
    //     columns: ["日期", "访问用户", "下单用户"],
    //     rows: [
    //       { 日期: "1/1", 访问用户: 1393, 下单用户: 1093, 下单率: 0.32 },
    //       { 日期: "1/2", 访问用户: 3530, 下单用户: 3230, 下单率: 0.26 },
    //       { 日期: "1/3", 访问用户: 2923, 下单用户: 2623, 下单率: 0.76 },
    //       { 日期: "1/4", 访问用户: 1723, 下单用户: 1423, 下单率: 0.49 },
    //       { 日期: "1/5", 访问用户: 3792, 下单用户: 3492, 下单率: 0.323 },
    //       { 日期: "1/6", 访问用户: 4593, 下单用户: 4293, 下单率: 0.78 }
    //     ]
    //   },
    //   settings: {
    //     metrics: ["访问用户", "下单用户"],
    //     dimension: ["日期"]
    //   }
    // },
    // {
    //   name: "自定义tooltip的显示",
    //   data: {
    //     columns: ["日期", "访问用户", "下单用户"],
    //     rows: [
    //       { 日期: "1/1", 访问用户: 1393, 下单用户: 1093, 下单率: 0.32 },
    //       { 日期: "1/2", 访问用户: 3530, 下单用户: 3230, 下单率: 0.26 },
    //       { 日期: "1/3", 访问用户: 2923, 下单用户: 2623, 下单率: 0.76 },
    //       { 日期: "1/4", 访问用户: 1723, 下单用户: 1423, 下单率: 0.49 },
    //       { 日期: "1/5", 访问用户: 3792, 下单用户: 3492, 下单率: 0.323 },
    //       { 日期: "1/6", 访问用户: 4593, 下单用户: 4293, 下单率: 0.78 }
    //     ]
    //   },
    //   settings: {
    //     tooltipMap: { 下单率: "我的下单率" }
    //   }
    // },
    // {
    //   name: "设置双 y 轴",
    //   data: {
    //     columns: ["日期", "访问用户", "下单用户", "下单率"],
    //     rows: [
    //       { 日期: "1/1", 访问用户: 1393, 下单用户: 1093, 下单率: 0.32 },
    //       { 日期: "1/2", 访问用户: 3530, 下单用户: 3230, 下单率: 0.26 },
    //       { 日期: "1/3", 访问用户: 2923, 下单用户: 2623, 下单率: 0.76 },
    //       { 日期: "1/4", 访问用户: 1723, 下单用户: 1423, 下单率: 0.49 },
    //       { 日期: "1/5", 访问用户: 3792, 下单用户: 3492, 下单率: 0.323 },
    //       { 日期: "1/6", 访问用户: 4593, 下单用户: 4293, 下单率: 0.78 }
    //     ]
    //   },
    //   settings: {
    //     axisSite: { right: ["下单率"] },
    //     yAxisType: ["KMB", "percent"],
    //     yAxisName: ["数值", "比率"],
    //     splitNumber: 8
    //   }
    // },
    {
      name: "设置指标名称",
      data: {
        columns: ["date", "balance", "age"],
        rows: [
          { date: "1-1", balance: 123, age: 3 },
          { date: "1-2", balance: 1223, age: 6 },
          { date: "1-3", balance: 2123, age: 9 },
          { date: "1-4", balance: 4123, age: 12 },
          { date: "1-5", balance: 3123, age: 15 },
          { date: "1-6", balance: 7123, age: 20 }
        ]
      },
      settings: {
        labelMap: {
          date: "日期",
          balance: "余额",
          age: "年龄"
        },
        axisSite: { right: ["age"] }
      }
    }
    // {
    //   name: "设置legend名称",
    //   data: {
    //     columns: ["日期", "余额", "年龄"],
    //     rows: [
    //       { 日期: "1-1", 余额: 123, 年龄: 3 },
    //       { 日期: "1-2", 余额: 1223, 年龄: 6 },
    //       { 日期: "1-3", 余额: 2123, 年龄: 9 },
    //       { 日期: "1-4", 余额: 4123, 年龄: 12 },
    //       { 日期: "1-5", 余额: 3123, 年龄: 15 },
    //       { 日期: "1-6", 余额: 7123, 年龄: 20 }
    //     ]
    //   },
    //   settings: {
    //     legendName: {
    //       余额: "remain"
    //     }
    //   }
    // },
    // {
    //   name: "带有较小百分比数值",
    //   data: {
    //     columns: ["日期", "比率"],
    //     rows: [
    //       { 日期: "1-1", 余额: 123, 比率: 0.00001 },
    //       { 日期: "1-2", 余额: 1223, 比率: 0.00002 },
    //       { 日期: "1-3", 余额: 2123, 比率: 0.00003 },
    //       { 日期: "1-4", 余额: 4123, 比率: 0.00007 },
    //       { 日期: "1-5", 余额: 3123, 比率: 0.00001 },
    //       { 日期: "1-6", 余额: 7123, 比率: 0.00003 }
    //     ]
    //   },
    //   settings: {
    //     yAxisType: ["percent"],
    //     digit: 4
    //   }
    // },
    // {
    //   name: "坐标轴配置",
    //   data: {
    //     columns: ["日期", "余额", "比率"],
    //     rows: [
    //       { 日期: "1-1", 余额: 123, 比率: 0.3 },
    //       { 日期: "1-2", 余额: 1223, 比率: 0.6 },
    //       { 日期: "1-3", 余额: 2123, 比率: 0.9 },
    //       { 日期: "1-4", 余额: 4123, 比率: 0.12 },
    //       { 日期: "1-5", 余额: 3123, 比率: 0.15 },
    //       { 日期: "1-6", 余额: 7123, 比率: 0.2 }
    //     ]
    //   },
    //   settings: {
    //     axisSite: {
    //       right: ["比率"]
    //     },
    //     yAxisType: ["KMB", "percent"]
    //   }
    // },
    // {
    //   name: "指标维度配置",
    //   data: {
    //     columns: ["日期", "余额", "比率"],
    //     rows: [
    //       { 日期: "1-1", 余额: 123, 比率: 0.1 },
    //       { 日期: "1-2", 余额: 1223, 比率: 0.2 },
    //       { 日期: "1-3", 余额: 2123, 比率: 0.3 },
    //       { 日期: "1-4", 余额: 4123, 比率: 0.4 },
    //       { 日期: "1-5", 余额: 3123, 比率: 0.5 },
    //       { 日期: "1-6", 余额: 7123, 比率: 0.6 }
    //     ]
    //   },
    //   settings: {
    //     dimension: ["比率"],
    //     metrics: ["余额"]
    //   }
    // },
    // {
    //   name: "坐标轴缩放配置",
    //   data: {
    //     columns: ["日期", "余额", "比率"],
    //     rows: [
    //       { 日期: "1-1", 余额: 1232, 比率: 0.1 },
    //       { 日期: "1-2", 余额: 1223, 比率: 0.2 },
    //       { 日期: "1-3", 余额: 2123, 比率: 0.3 },
    //       { 日期: "1-4", 余额: 4123, 比率: 0.4 },
    //       { 日期: "1-5", 余额: 3123, 比率: 0.5 },
    //       { 日期: "1-6", 余额: 7123, 比率: 0.6 }
    //     ]
    //   },
    //   settings: {
    //     dimension: ["比率"],
    //     metrics: ["余额"],
    //     scale: [true, true]
    //   }
    // },
    // {
    //   name: "坐标轴值域配置",
    //   data: {
    //     columns: ["日期", "余额", "比率"],
    //     rows: [
    //       { 日期: "1-1", 余额: 1232, 比率: 0.1 },
    //       { 日期: "1-2", 余额: 1223, 比率: 0.2 },
    //       { 日期: "1-3", 余额: 2123, 比率: 0.3 },
    //       { 日期: "1-4", 余额: 4123, 比率: 0.4 },
    //       { 日期: "1-5", 余额: 3123, 比率: 0.5 },
    //       { 日期: "1-6", 余额: 7123, 比率: 0.6 }
    //     ]
    //   },
    //   settings: {
    //     dimension: ["比率"],
    //     metrics: ["余额"],
    //     min: [1000],
    //     max: [5000]
    //   }
    // },
    // {
    //   name: "面积图",
    //   data: {
    //     columns: ["日期", "余额", "比率"],
    //     rows: [
    //       { 日期: "1-1", 余额: 123, 比率: 0.1 },
    //       { 日期: "1-2", 余额: 1223, 比率: 0.2 },
    //       { 日期: "1-3", 余额: 2123, 比率: 0.3 },
    //       { 日期: "1-4", 余额: 4123, 比率: 0.4 },
    //       { 日期: "1-5", 余额: 3123, 比率: 0.5 },
    //       { 日期: "1-6", 余额: 7123, 比率: 0.6 }
    //     ]
    //   },
    //   settings: {
    //     area: true
    //   }
    // },
    // {
    //   name: "堆叠折线图",
    //   data: {
    //     columns: ["日期", "2015", "2016"],
    //     rows: [
    //       { 日期: "1-1", "2015": 100, "2016": 40 },
    //       { 日期: "1-2", "2015": 110, "2016": 60 },
    //       { 日期: "1-3", "2015": 140, "2016": 10 },
    //       { 日期: "1-4", "2015": 260, "2016": 30 },
    //       { 日期: "1-5", "2015": 300, "2016": 20 },
    //       { 日期: "1-6", "2015": 380, "2016": 70 }
    //     ]
    //   },
    //   settings: {
    //     stack: {
    //       年份: ["2015", "2016"]
    //     }
    //   }
    // }
  ]
};
